tests/python_package_test/test_sequence.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python
+# -*- coding:utf-8 -*-
+# Author: mithril
+# Created Date: 2024-07-14 14:18:46
+# Last Modified: 2024-07-14 14:44:50
+
+
+import numpy as np
+import pytest
+import sklearn.datasets
+import lightgbm as lgb
+
+
+class PartitionSequence(lgb.Sequence):
+    def __init__(self, data:np.ndarray, batch_size=4096):
+        self.data = data
+        self.batch_size = batch_size
+
+    def __getitem__(self, idx):
+        return self.data[idx]
+
+    def __len__(self):
+        return len(self.data)
+
+
+def test_list_of_sequence():
+    X, y = sklearn.datasets.load_breast_cancer(return_X_y=True)
```

Comment:
When you move this test, please use the cached version of this function.

https://github.com/microsoft/LightGBM/blob/830763d9e58e05b678fb88e9b6f574a20158a36a/tests/python_package_test/test_basic.py#L221

https://github.com/microsoft/LightGBM/blob/830763d9e58e05b678fb88e9b6f574a20158a36a/tests/python_package_test/test_basic.py#L221

That'll make the tests a little faster.